src/fish_tests.cpp

```diff
@@ -1821,12 +1821,12 @@ class test_lru_t : public lru_cache_t<test_lru_t, int> {
 
     std::vector<value_type> evicted;
 
-    void entry_was_evicted(const wcstring &key, int val) { evicted.push_back({key, val}); }
+    void entry_was_evicted(const wcstring &key, int val) { evicted.emplace_back(key, val); }
 
     std::vector<value_type> values() const {
         std::vector<value_type> result;
         for (auto p : *this) {
-            result.push_back(p);
+            result.emplace_back(p);
```

Comment:
I think if we use `const auto &p` in the loop, then we can elide the move as well (and keep using `push_back`).

Reply:
Didn't make a difference. clang-tidy still converted to emplace_back.

This loop could also be converted to use std::copy which internally uses push_back.

Reply:
Interesting!